python/xir/program.py

```diff
@@ -567,6 +567,65 @@ def serialize(self, minimize: bool = False) -> str:
             return strip(res_script)
         return res_script
 
+    @staticmethod
+    def merge(*programs) -> "XIRProgram":
```

Comment:
```suggestion
    def merge(*programs) -> XIRProgram:
```
Could this be the class itself?

Reply:
I wish!  Unfortunately, until we upgrade to Python 3.10+, we need to use the string representation of the class as per [PEP 484](https://www.python.org/dev/peps/pep-0484/#forward-references).  Alternatively, we can just put
```python
from __future__ import annotations
```
at the top of the file.